api/client/commit.go

```diff
@@ -67,6 +67,9 @@ func (cli *DockerCli) CmdCommit(args ...string) error {
 		}
 	}
 	stream, _, _, err := cli.call("POST", "/commit?"+v.Encode(), config, nil)
+	if stream != nil {
+		defer stream.Close()
+	}
```

Comment:
Can you add this `defer` after the err check so we'll be sure stream is not nil?